src/client/opamAction.ml

```diff
@@ -366,15 +366,27 @@ let prepare_package_source st nv dir =
   in
   let check_extra_files =
     try
-      List.iter (fun (src,base,hash) ->
-          if not (OpamHash.check_file (OpamFilename.to_string src) hash) then
-            failwith
-              (Printf.sprintf "Bad hash for %s" (OpamFilename.to_string src))
+      let extra_files_dir =
+        OpamPath.Switch.extra_files_dir st.switch_global.root st.switch
+      in
+      List.iter (fun (base, hash) ->
+          let dst = OpamFilename.create dir base in
```

Comment:
If the check fails, `dir/base` is created and not deleted. Better create it at last moment.